modin/pandas/series.py

```diff
@@ -976,6 +976,31 @@ def slice_shift(self, periods=1, axis=0):
                 )
             )
 
+    def unstack(self, level=-1, fill_value=None):
+        """
+        Unstack, a.k.a. pivot, Series with MultiIndex to produce DataFrame.
+        The level involved will automatically get sorted.
+        Parameters
+        ----------
+        level : int, str, or list of these, default last level
+            Level(s) to unstack, can pass level name.
+        fill_value : scalar value, default None
+            Value to use when replacing NaN values.
+        Returns
+        -------
+        DataFrame
+            Unstacked Series.
+        """
+        from .dataframe import DataFrame
+
+        return DataFrame(
+            query_compiler=self._query_compiler.unstack(
+                map_func=lambda df: pandas.DataFrame(
+                    df.squeeze(axis=1).unstack(level=level, fill_value=fill_value)
+                ),
+            )
+        )
```

Comment:
I suppose that `query_compiler::unstack` should repeat the interface of `DataFrame::unstack`, because `query_compiler` is backend-specific. Let's say another non-pandas backed already have `unstack` implementation which expects `level` and `fill_value` params, but in that implementation, it will get some python callable (which actually could be not supported by non-pandas backends).

I've read your comment about the reasons you've implemented it this way, but do the results differ if we call `unstack` on `DataFrame` and `DataFrame.squeeze()`? If it is, can we translate the result of `unstack` from `DataFrame_which_is_actually_series` to the correct answer? If no, maybe we should make two different implementations of `unstack` in `query_compiler` for `Series` and for `DataFrame`. @devin-petersohn, need your advice.

Reply:
Yes, I agree with @dchigarev. If possible, these details like `map_func` should be kept in the query compiler because other engines may not be able to apply them.

`unstack` is `pivot` on a level of the index instead of a column. `pivot` and `unstack` are both a combination of groupby and transpose. It might be better to try to implement them this way.

Reply:
I tried to fix my implementation using `apply_full_axis`. But I get data with incorrect size of output when result is `Series`. In reproducer below results should have 1089 rows, but I get only 1056 rows. Missed 33 rows are 33rd column of original `DataFrame`. There are only two partitions (size of each is 528 rows) internally (in [line](https://github.com/modin-project/modin/blob/master/modin/engines/base/frame/partition_manager.py#L186) ). Should there be third partition in this case, which will be consist 33rd column of original `DataFrame`? @devin-petersohn Have you encountered such problem?

Code of test:
```
def test_unstack(self):
        data = {f"Col{i}": np.arange(33) for i in range(33)}
        modin_df = pd.DataFrame(data)
        res_p = pandas_df.unstack(level=0)
        res_m = modin_df.unstack(level=0)

        p_idx = res_p.index
        m_idx = res_m.index

        i = 0
        for idx in len(p_idx):
            print(idx, "_", m_idx[i])
            i = i +1
``` 


Also I don't properly understand why `ustack` is combination `groupby` and `transpose`. Can you explain what you mean?